crates/configuration/src/network.rs

```diff
@@ -0,0 +1,92 @@
+use crate::{
+    shared::types::{IpAddress, MultiAddress, Timeout},
+    HrmpChannelConfig, ParachainConfig, RelaychainConfig,
+};
+
+/// Global settings applied to an entire network.
+pub struct GlobalSettings {
```

Comment:
This does not have to be public (at least as of this code):
```suggestion
struct GlobalSettings {
```

Reply:
We can't else are exposing the private API in the following method:

```rust
    pub fn with_global_settings(self, f: fn(GlobalSettings) -> GlobalSettings) -> Self {
        Self {
            global_settings: f(self.global_settings),
            ..self
        }
    }
```

Reply:
Im not sure I understand. This method exists in the same file.
What I am trying to say, is that everything should be init as private until they are needed to be declared as public. Its better than declare everything by default as public

Reply:
The method exists in the same file but is public, this is the API we will expose to the user. Every struct field is private but every method on the builder is public else the user won't be able to fill some fields due to the field being private.